### PR TITLE
add iOS numeric http header support

### PR DESCRIPTION
**Pull request by trave7er**
NSMutableURLRequest's allHTTPHeaderFields property does not support value other than NSString as of today. The argument is a NSDictionary<NSString *, id> *, so a conversion to NSString of all values is necessary.

The issue does not show up when using fetch api, but does show up when using browserified superagent.

### CODE REVIEW

**Comment by trave7er** [comment]
This PR is addressed to issue https://github.com/facebook/react-native/issues/4715

**Comment by nicklockwood** [comment]
I see this as a type error, in which case we should probably detect it and display an error rather than silently converting it, as the conversion is quite expensive.

File: Libraries/Network/RCTNetworking.m [hunk 1]
```diff
@@ -209,7 +209,10 @@ - (RCTURLRequestCancellationBlock)buildRequest:(NSDictionary<NSString *, id> *)q
   NSURL *URL = [RCTConvert NSURL:query[@"url"]]; // this is marked as nullable in JS, but should not be null
   NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:URL];
   request.HTTPMethod = [RCTConvert NSString:RCTNilIfNull(query[@"method"])].uppercaseString ?: @"GET";
-  request.allHTTPHeaderFields = [RCTConvert NSDictionary:query[@"headers"]];
+  NSMutableDictionary *headers = [[RCTConvert NSDictionary:query[@"headers"]] mutableCopy];
+  [headers enumerateKeysAndObjectsUsingBlock:^(NSString *  _Nonnull key, id  _Nonnull obj, BOOL * _Nonnull stop) {
+    [headers setValue:[NSString stringWithFormat:@"%@", obj] forKey:key];
```
Comment (nicklockwood) [inline Libraries/Network/RCTNetworking.m]: A couple of issues with this:

1) `setValue:forKey:` is actually part of the KVC (Key Value Coding) protocol, and should not be used for setting dictionary keys. The method you probably meant was `setObject:forKey:`, but we prefer to use `dict[key] = value` syntax anyway. I'd write it as:

    headers[key] = [value description];

2) Are you sure it's safe to mutate the `headers` dictionary while you are enumerating it? That normally crashes with a "Collection was mutated while being enumerated." exception.

**Comment by trave7er** [comment]
hi @nicklockwood , sorry for the mistake. The initial commit was a quick fix right before lunch. :grin:

**Comment by trave7er** [comment]
hi @nicklockwood , let me know if there is something you want to merge into the codebase. Thanks

**Comment by facebook-github-bot-6** [comment]
@nicklockwood would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

**Comment by javache** [comment]
Per Nick's comment, this doesn't seem like something we'd want to merge. Feel free to submit a new PR with alternative solutions.

**Pull request closed without merging**